text/0000-crates.io-default-ranking.md

```diff
@@ -0,0 +1,707 @@
+- Number of lines in the README file, if specified in Cargo.toml
+- Number of lines in Rust files: `find . -name '*.rs' | xargs wc -l`
+
+We would then add the lines in the README to the lines of documentation and
+subtract the lines of documentation from the total lines of code in order to
+get the ratio of documentation to code. Test code (and any documentation within
+test code) *is* part of this calculation.
+
+Any crate getting in the top 20% of all crates would get a badge saying "well
+documented".
+
+Additionally, lists of crates would have a badge showing the number of files in
+the standard `/examples` directory, if any. A further enhancement would be to
+make that badge link to the examples displayed somewhere (crates.io? in the
+repository? in the documentation?).
+
+* combine:
+  * 1,195 lines of documentation
+  * 99 lines in README.md
+  * 5,815 lines of Rust
+  * (1195 + 99) / (5815 - 1195) = 1294/4620 = .28
+
+* nom:
+  * 2,263 lines of documentation
+  * 372 lines in README.md
+  * 15,661 lines of Rust
+  * (2263 + 372) / (15661 - 2263) = 2635/13398 = .20
+
+* peresil:
+  * 159 lines of documentation
+  * 20 lines in README.md
+  * 1,341 lines of Rust
+  * (159 + 20) / (1341 - 159) = 179/1182 = .15
+
+* lalrpop: ([in the /lalrpop directory in the repo][lalrpop-repo])
+  * 742 lines of documentation
+  * 110 lines in ../README.md
+  * 94,104 lines of Rust
+  * (742 + 110) / (94104 - 742) = 852/93362 = .01
+
+* peg:
+  * 3 lines of documentation
+  * no readme specified in Cargo.toml
+  * 1,531 lines of Rust
+  * (3 + 0) / (1531 - 3) = 3/1528 = .00
+
+[lalrpop-repo]: https://github.com/nikomatsakis/lalrpop/tree/master/lalrpop
+
+If we assume these are all the crates on crates.io for this example, then
+combine is the top 20% and would get a badge. None of the crates have files in
+`/examples`, so none would have the examples badge.
+
+### Maintenance
+
+We can add an optional attribute to Cargo.toml that crate authors could use to
+self-report their maintenance intentions. The valid values would be along the
+lines of the following, and would influence the ranking in the order they're
+presented:
+
+- **Actively developed**, meaning new features are being added and bugs are
+  being fixed
+- **Passively maintained**, meaning there are no plans for new features, but
+  the maintainer intends to respond to issues that get filed
+- **As-is**, meaning the crate is feature complete, the maintainer does not
+  intend to continue working on it or providing support, but it works for the
+  purposes it was designed for
+- None, we don't display anything, since the maintainer has not chosen to
+  specify their intentions, potential crate users will need to investigate on
+  their own
+- **Experimental**, meaning the author wants to share it with the community but
+  is not intending to meet anyone's particular use case
+- **Looking for maintainer**, meaning the current maintainer would like to give
```

Comment:
Unrelated to the goals of this rfc, I think it would be neat to see a "crate adoption agency" section of crate.io for finding loving parents and warm homes for these :)